src/Query/index.ts

```diff
@@ -149,11 +150,22 @@ export class Query {
       );
     }
 
+    // if user requested searchProperty, must match this property name
+    // TODO: change to filters.
+    if (searchProperty !== null && searchProperty.length !== 0) {
+      compound.must.push({
+        phrase: {
+          path: 'searchProperty',
+          query: searchProperty,
```

Comment:
we can do an if/else here. the must conditions always must apply so they will be the same res.

on the other hand, it might be a good idea to do one filter in case this creates some minimal double query filtering on db aggregation side